Allow to provide plan via stdin

JeanFred opened the pull request:
Currently, prettyplan-cli takes it upon himself to obtain
the plan to prettify by running `terraform plan` by itself.

However, there can be cases where this not an option:
 - `terraform` is not readily available in the PATH
 - Additional flags need to be passed to terraform
 - Terraform is invoked by a wrapper (such as terragrunt)

This adds an additional flag, `input`.
If set, `prettyplan-cli` expects the plan to be provided via stdin.

Resolves #9

JeanFred commented:
I went ahead and gave this a try − the result might not be the most idiomatic go code. Happy to make changes :)

chrislewisdev commented on main.go:
```diff
@@ -18,13 +19,14 @@ import (
 func main() {
 	openFileFlag := flag.Bool("open", false, "To open the HTML report once generated")
 	fileNameFlag := flag.String("filename", "prettyplan-output.html", "Specify a filename other than prettyplan-output.html")
+	inputFlag := flag.Bool("input", false, "Whether to read the plan from stdin instead of running terraform")
```
The name "input" is quite generic, I think `stdin` would at least be a more fit-for-purpose name for the flag here:

```suggestion
	inputFlag := flag.Bool("stdin", false, "Whether to read the plan from stdin instead of running terraform")
```

JeanFred replied:
Agreed, much better name.

chrislewisdev reviewed (commented):
Thanks for the PR Jean, this looks quite reasonable to me - just one suggestion re: naming before we merge this in :)

(And apologies for the delay in responding to your submissions - as you might have gathered, I haven't touched this in a while and have not been checking Github very often)

JeanFred commented on main.go:
```diff
@@ -79,15 +81,30 @@ type planInfo struct {
+	var rawPlan []byte
+	var err error
+	if stdinFlag {
+		fmt.Println("Reading plan from stdin...")
```
Is this actually a good idea to print this?

JeanFred commented:
> Thanks for the PR Jean, this looks quite reasonable to me - just one suggestion re: naming before we merge this in :)

Good suggestion, done.

> (And apologies for the delay in responding to your submissions - as you might have gathered, I haven't touched this in a while and have not been checking Github very often)

No problem at all, of course :) Thanks for taking the time to review this!

JeanFred committed:
Allow to provide plan via stdin

Currently, prettyplan-cli takes it upon himself to obtain
the plan to prettify by running `terraform plan` by itself.

However, there can be cases where this not an option:
 - `terraform` is not readily available in the PATH
 - Additional flags need to be passed to terraform
 - Terraform is invoked by a wrapper (such as terragrunt)

This adds an additional flag, `stdin`.
If set, `prettyplan-cli` expects the plan to be provided via stdin.

Resolves #9

JeanFred committed:
Extract method getRawPlan from getPlan

This paves the way to getting the raw plan in other ways
than executing `terraform plan` (for example through stdin).

chrislewisdev reviewed (approved):
Thanks for the PR Jean, I'll go ahead and merge this in and tag up a new release.
